src/storage/manager.rs

```diff
@@ -108,9 +109,18 @@ impl StateManager {
     pub fn current_status(&self) -> Status {
         let status = self.state.lock().status();
         let stored_bytes = get_directory_size(&self.fs.root);
+        let datasets_index = self.datasets_index.lock();
+        let mut unavailability_map: Vec<u8> = Vec::default();
+        unavailability_map.resize(datasets_index.chunks_ordinals_map.len(), 1);
+        for chunk_ref in &status.available {
+            if let Some(ordinal) = datasets_index.chunks_ordinals_map.get(&chunk_ref.chunk) {
+                unavailability_map[*ordinal as usize] = 0
+            }
+        }
         Status {
             available: to_ranges(status.available),
             downloading: to_ranges(status.downloading),
```

Comment:
Is it still used? `to_ranges` is a slow operation, so if not, please remove it

Reply:
In HTTPServer